### ISSUE TITLE
False negative for `Lint/Void` with ternary expression 

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13104]
## Expected behavior

Offense for the following:

```rb
foo? ? bar : 1
foo? ? 1 : bar
```

## Actual behavior

No offense. There is already handling for `1 if foo?` so this seems like a logical improvement.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.65.1 (using Prism 0.30.0, rubocop-ast 1.31.3, running on ruby 3.3.2) [aarch64-linux-android]
```

**Comment by vlad-pisanov** [comment]
That's an interesting thought. Expanding on it, any `if`/`case` statement in a void context that returns a literal (in _any_ branch!) should potentially be flagged by this cop.

```ruby
if cond?
  42 # literal returned into void context ⚠️ 
end
```

```ruby
if cond?
  42 # literal returned into void context ⚠️ 
else
  foo
end
```

```ruby
case val
when 'foo' then 42 # literal returned into void context ⚠️ 
end
```

**Comment by Earlopain** [comment]
When you write it like that, it looks very similar to #12393
